src/plugins/data/public/search/sync_search_strategy.ts

```diff
@@ -47,7 +47,7 @@ export const syncSearchStrategyProvider: TSearchStrategyProvider<typeof SYNC_SEA
       signal: options.signal,
     });
 
-    response.then(() => loadingCount$.next(loadingCount$.getValue() - 1));
+    response.finally(() => loadingCount$.next(loadingCount$.getValue() - 1));
```

Comment:
Curious, if this is a bug fix and if there should be a test case for this?

Reply:
Yes it is and yes there should be.

Reply:
Added in 159f5f04b68010d23784175ae34b28eaf4b02196.